### DISCUSSION TITLE
Execute Kohya SS without Webui

### DISCUSSION THREAD

**Discussion by zhao-binxiang** [kohya-ss/sd-scripts#747] [Q&A]
For the Lora tab training, how do I execute a specific .py file with parameters instead of using the kohya ss webui? In another words, instead of selecting the parameters in the webui, how could I just run some training_file.py for Lora training?

**Comment by araleza** [comment]
Just click 'Print training command' at the bottom of the webui, and have a look in your command line window.  It'll print a large command there that you can copy and paste to run later to get the same result as the webui.

But before you run the command, you have to activate the python environment that the webui uses.  So go to the webui directory in your command line and type 'source venv/bin/activate' on Linux, or 'venv\Scripts\activate.bat' on Windows.

**Comment by gahoccode** [comment]
> Just click 'Print training command' at the bottom of the webui, and have a look in your command line window.  It'll print a large command there that you can copy and paste to run later to get the same result as the webui.
> 
> 
> 
> But before you run the command, you have to activate the python environment that the webui uses.  So go to the webui directory in your command line and type 'source venv/bin/activate' on Linux, or 'venv\Scripts\activate.bat' on Windows.

Hi could you elaborate on which python file I should run? I'm assuming the flux_train.py in the sd scripts folder? Something like python flux_train.py config.json?

**Reply by MNeMoNiCuZ** [reply]
> > Just click 'Print training command' at the bottom of the webui, and have a look in your command line window.  It'll print a large command there that you can copy and paste to run later to get the same result as the webui.
> > But before you run the command, you have to activate the python environment that the webui uses.  So go to the webui directory in your command line and type 'source venv/bin/activate' on Linux, or 'venv\Scripts\activate.bat' on Windows.
> 
> Hi could you elaborate on which python file I should run? I'm assuming the flux_train.py in the sd scripts folder? Something like python flux_train.py config.json?

It seems to be `flux_train_network.py`.

This guide might be able to help:
https://civitai.com/articles/7365/flux-lora-training-low-gpu-kohya-ssdscript-linux

It got me through setting up the config for the first time.

**Reply by gahoccode** [reply]
> > > Just click 'Print training command' at the bottom of the webui, and have a look in your command line window.  It'll print a large command there that you can copy and paste to run later to get the same result as the webui.
> > > But before you run the command, you have to activate the python environment that the webui uses.  So go to the webui directory in your command line and type 'source venv/bin/activate' on Linux, or 'venv\Scripts\activate.bat' on Windows.
> > 
> > 
> > Hi could you elaborate on which python file I should run? I'm assuming the flux_train.py in the sd scripts folder? Something like python flux_train.py config.json?
> 
> It seems to be `flux_train_network.py`.
> 
> This guide might be able to help: https://civitai.com/articles/7365/flux-lora-training-low-gpu-kohya-ssdscript-linux
> 
> It got me through setting up the config for the first time.

Oh yeah thanks this works. I figured out a better way to do it though. It's something like this accelerate launch --num_cpu_threads_per_process=2 "flux_train_network.py" --config_file=..\config_lora.toml (the toml file generated using the GUI can be passed into this command line to run directly)
